diff --git a/media/libstagefright/omx/Android.bp b/media/libstagefright/omx/Android.bp
index 7a7bcd3c1d..8b8cefe6ea 100644
--- a/media/libstagefright/omx/Android.bp
+++ b/media/libstagefright/omx/Android.bp
@@ -196,7 +196,7 @@ cc_defaults {
         cfi: true,
     },
 
-    compile_multilib: "32",
+    compile_multilib: "first",
 }
 
 cc_library_shared {
diff --git a/services/mediacodec/Android.bp b/services/mediacodec/Android.bp
index c4efbaa141..cf353e2789 100644
--- a/services/mediacodec/Android.bp
+++ b/services/mediacodec/Android.bp
@@ -114,7 +114,7 @@ cc_binary {
     ],
 
     // OMX interfaces force this to stay in 32-bit mode;
-    compile_multilib: "32",
+    compile_multilib: "first",
 
     init_rc: ["android.hardware.media.omx@1.0-service.rc"],
 
diff --git a/services/mediacodec/seccomp_policy/mediacodec-arm64.policy b/services/mediacodec/seccomp_policy/mediacodec-arm64.policy
index d2baa7f279..7f7f13c300 100644
--- a/services/mediacodec/seccomp_policy/mediacodec-arm64.policy
+++ b/services/mediacodec/seccomp_policy/mediacodec-arm64.policy
@@ -12,6 +12,8 @@ writev: 1
 dup: 1
 ppoll: 1
 mmap2: 1
+fstat:1
+mmap:1
 getrandom: 1
 memfd_create: 1
 ftruncate: 1
@@ -34,6 +36,8 @@ setpriority: 1
 getuid32: 1
 fstat64: 1
 fstatfs64: 1
+fstatfs: 1
+newfstatat: 1
 pread64: 1
 faccessat: 1
 readlinkat: 1
